### DISCUSSION TITLE
Can the URL be translated?

### DISCUSSION THREAD

**Discussion by aljawaid** [aljawaid/ColorManager#11] [Q&A] [answered]
https://github.com/aljawaid/ColorManager/blob/b980f3e0d60037b3d9bd27692274a614d9ee62db/Plugin.php#L25

@creecros 

Can it be `...com/settings/t('colors')` ?

**Comment by aljawaid** [comment]
it seems not, I tried the above... also this
```
$url = $this->request->getStringParam(t('color-manager'));
        $this->route->addRoute('/settings/:url', 'ColorManagerController', 'show', 'ColorManager');
```
but it didnt work

**Comment by creecros** [comment]
are you good here? not sure what you are trying to accomplish, but `$url = $this->request->getStringParam(t('color-manager'));` is not how you set the `:url` variable. that needs to come from the controller function that will ultimately send the redirect to the function `show`.

something like:
```
$url = 'whatever you want it to equal';
$this->response->redirect($this->helper->url->to('ColorManagerController', 'show', array('plugin' => 'ColorManager', 'url' => $url)), true);
```

**Reply by aljawaid** [reply]
After reading it 4 times I think I see what you mean and it may be possible to have a translated word in the URL.

I will look at it tonight when I get home.

**Reply by creecros** [reply]
https://github.com/aljawaid/ColorManager/commit/46ef02dfae7aaa0c681b3ebd3511fa4185fa304a

**Reply by creecros** [reply]
I didn't check anywhere else, but basically, do what i did in that commit, everywhere you call the `show` function in that controller.

**Reply by aljawaid** [reply]
> I didn't check anywhere else, but basically, do what i did in that commit, everywhere you call the `show` function in that controller. 

Ah OK great. That's the only place it's needed but I will check.

You did it in the sidebar, interesting, I was going to do it in the actual controller tonight. It makes sense what you have done. Thanks.

**Reply by creecros** [reply]
well, it should be needed in the controller where it redirects back to `show`, example:
https://github.com/aljawaid/ColorManager/blob/46ef02dfae7aaa0c681b3ebd3511fa4185fa304a/Controller/ColorManagerController.php#L67

**Reply by aljawaid** [reply]
> well, it should be needed in the controller where it redirects back to `show`, example:
> 
> https://github.com/aljawaid/ColorManager/blob/46ef02dfae7aaa0c681b3ebd3511fa4185fa304a/Controller/ColorManagerController.php#L67

done in f7b5ef3

@creecros but should I add it also to:
https://github.com/aljawaid/ColorManager/blob/f7b5ef31472c7d75b0456a4590af4da1eeada12d/Controller/ColorManagerController.php#L29

**Reply by creecros** [reply]
nope

**Comment by aljawaid** [comment]
I left it as it got too complicated....

I just wanted the url to be able to be lang-specific... so `/color-manager` for default but `/colour-manager` for english UK

**Comment by aljawaid** [comment, marked as answer]
**NO**

Whilst the translation does work here https://github.com/aljawaid/ColorManager/discussions/11#discussioncomment-4503738 it seems to cause issues with other plugins which have pages in settings. By translating the url, ColorManager was overriding the rendering/display of settings pages of other plugins so the feature had to be reverted.
